src/js/letters/components/Address.jsx

```diff
@@ -69,75 +61,104 @@ class Address extends React.Component {
   }
 
   render() {
+    const addressType = this.props.value.type;
+
+    // Depending on the type of address, the fields for state and city
+    // are called different things. Set new variables for the name of
+    // the appropriate city and state field, and use those below for
+    // the field components.
+    let cityField;
+    let stateField;
+    if (addressType === 'MILITARY') {
+      cityField = 'militaryPostOfficeTypeCode';
+      stateField = 'militaryStateCode';
+    } else {
+      cityField = 'city';
+      stateField = 'state';
+    }
+
+    // Our hard-coded list of countries has the value for the U.S. as
+    // 'USA', but the value sent to us from EVSS is 'US'. This will cause
+    // the field to not populate correctly.
+    // This will be changed once we pull the real country list from the
+    // address endpoint.
+    let selectedCountry;
+    if (this.props.value.country === undefined && addressType === 'MILITARY') {
+      selectedCountry = 'USA';
+    } else if (this.props.value.country === 'US') {
+      selectedCountry = 'USA';
+    } else {
+      selectedCountry = this.props.value.country;
+    }
+
     let stateList = [];
-    const selectedCountry = this.props.value.country === 'US' ? 'USA' : this.props.value.country;
     if (states[selectedCountry]) {
       stateList = states[selectedCountry];
       if (this.props.value.city && this.isMilitaryCity(this.props.value.city)) {
         stateList = stateList.filter(state => state.value === 'AE' || state.value === 'AP' || state.value === 'AA');
       }
     }
 
-    const stateProvince = _.hasIn(states, this.props.value.country === 'US' ? 'USA' : this.props.value.country)
-      ? (<ErrorableSelect errorMessage={this.isValidAddressField(this.props.value.state) ? undefined : 'Please enter a valid state/province'}
-        label={this.props.value.country === 'CAN' ? 'Province' : 'State'}
+    const stateProvince = _.hasIn(states, selectedCountry)
+      ? (<ErrorableSelect errorMessage={this.isValidAddressField(this.props.value[stateField]) ? undefined : 'Please enter a valid state/province'}
+        label="State"
```

Comment:
This can also go away (see same comment in the issue above): there shall be no province or state names for international addresses because the VA format says so.